src/wallet/spend.cpp

```diff
@@ -105,6 +105,47 @@ void CoinsResult::clear()
     other.clear();
 }
 
+// Fetch and validate the coin control selected inputs.
+// Coins could be internal (from the wallet) or external.
+util::Result<SelectedInputs> FetchSelectedInputs(const CWallet& wallet, const CCoinControl& coin_control,
```

Comment:
I'd recommend calling this `GetPreSelectedInputs`. we tend to use `Get*` throughout the codebase and since this function is for pre-selected inputs only, it makes sense to make the function name specific

Reply:
The rationale here to use `Fetch*` instead of `Get*` was that we aren't only accessing wallet's UTXOs. We are fetching external UTXO provided by the coin control object as well.

So, with the `Fetch*` prefix aimed to make that distinction clear.
If the function would be only getting wallet internal data, then yeah, this should had been a `Get*` for sure.